pennylane/operation.py

```diff
@@ -1385,6 +1385,56 @@ def compute_qfunc_decomposition(*args, **hyperparameters) -> None:
 
         raise DecompositionUndefinedError
 
+    @classproperty
+    def resource_keys(self) -> set:  # pylint: disable=no-self-use
```

Comment:
```suggestion
    def resource_keys(self) -> Union[set, frozenset]:  # pylint: disable=no-self-use
```
Might be good to allow it to be a frozenset as well.